avocado/core/job.py

```diff
@@ -483,7 +483,7 @@ def _run(self):
 
         if 'INTERRUPTED' in summary:
             self.exitcode |= exit_codes.AVOCADO_JOB_INTERRUPTED
-        if 'FAIL' in summary:
+        if 'FAIL' in summary and not getattr(self.args, 'relax_exit_code', False):
```

Comment:
Well this solution basically accepts 0 and 1 as a valid exit code, but this might not be enough. In jenkins I'm using shell job and I always ignore avocado exit code even on job interruption, errors and other failures, because even on job error I prefer per-test results and not just build failed result. I only get build failure when avocado does not produces the xunit output.

So I'm wondering whether we need something like `ignore_exit_code` pattern, or whether we want to implement `relax_exit_code` to ignore FAILs or to ignore all test statuses.

I'd prefer not doing anything as it works for :smile:, but as you claim there are jobs which can't be executed with shell I'd go with ignoring all return codes and only when really needed I'd go the bitmask way.